jax/_src/third_party/scipy/special.py

```diff
@@ -0,0 +1,223 @@
+     `here
+     <https://github.com/scipy/scipy/blob/v1.14.0/scipy/special/special/cephes/fresnl.h>`_.
+  """
+  if jnp.iscomplexobj(x):
+    raise NotImplementedError(
+        'Support for complex-valued inputs is not implemented yet.')
+
+  xxa, = promote_args_inexact("fresnel", x)
+
+  # This part is mostly a direct translation of SciPy's C++ code.
+
+  fresnl_sn = jnp.array([
+    -2.99181919401019853726e3,
+    +7.08840045257738576863e5,
+    -6.29741486205862506537e7,
+    +2.54890880573376359104e9,
+    -4.42979518059697779103e10,
+    +3.18016297876567817986e11,
+  ])
+
+  fresnl_sd = jnp.array([
+    +1.00000000000000000000e0,
+    +2.81376268889994315696e2,
+    +4.55847810806532581675e4,
+    +5.17343888770096400730e6,
+    +4.19320245898111231129e8,
+    +2.24411795645340920940e10,
+    +6.07366389490084639049e11,
+  ])
+
+  fresnl_cn = jnp.array([
+    -4.98843114573573548651e-8,
+    +9.50428062829859605134e-6,
+    -6.45191435683965050962e-4,
+    +1.88843319396703850064e-2,
+    -2.05525900955013891793e-1,
+    +9.99999999999999998822e-1,
+  ])
+
+  fresnl_cd = jnp.array([
+    +3.99982968972495980367e-12,
+    +9.15439215774657478799e-10,
+    +1.25001862479598821474e-7,
+    +1.22262789024179030997e-5,
+    +8.68029542941784300606e-4,
+    +4.12142090722199792936e-2,
+    +1.00000000000000000118e0,
+  ])
+
+  fresnl_fn = jnp.array([
+    +4.21543555043677546506e-1,
+    +1.43407919780758885261e-1,
+    +1.15220955073585758835e-2,
+    +3.45017939782574027900e-4,
+    +4.63613749287867322088e-6,
+    +3.05568983790257605827e-8,
+    +1.02304514164907233465e-10,
+    +1.72010743268161828879e-13,
+    +1.34283276233062758925e-16,
+    +3.76329711269987889006e-20,
+  ])
+
+  fresnl_fd = jnp.array([
+    +1.00000000000000000000e0,
+    +7.51586398353378947175e-1,
+    +1.16888925859191382142e-1,
+    +6.44051526508858611005e-3,
+    +1.55934409164153020873e-4,
+    +1.84627567348930545870e-6,
+    +1.12699224763999035261e-8,
+    +3.60140029589371370404e-11,
+    +5.88754533621578410010e-14,
+    +4.52001434074129701496e-17,
+    +1.25443237090011264384e-20,
+  ])
+
+  fresnl_gn = jnp.array([
+    +5.04442073643383265887e-1,
+    +1.97102833525523411709e-1,
+    +1.87648584092575249293e-2,
+    +6.84079380915393090172e-4,
+    +1.15138826111884280931e-5,
+    +9.82852443688422223854e-8,
+    +4.45344415861750144738e-10,
+    +1.08268041139020870318e-12,
+    +1.37555460633261799868e-15,
+    +8.36354435630677421531e-19,
+    +1.86958710162783235106e-22,
+  ])
+
+  fresnl_gd = jnp.array([
+    +1.00000000000000000000e0,
+    +1.47495759925128324529e0,
+    +3.37748989120019970451e-1,
+    +2.53603741420338795122e-2,
+    +8.14679107184306179049e-4,
+    +1.27545075667729118702e-5,
+    +1.04314589657571990585e-7,
+    +4.60680728146520428211e-10,
+    +1.10273215066240270757e-12,
+    +1.38796531259578871258e-15,
+    +8.39158816283118707363e-19,
+    +1.86958710162783236342e-22,
+  ])
+
+  @jit
+  def sincospi(
```

Comment:
you should define jit-compiled helper functions at the outer scope, otherwise this will be re-compiled each time `fresnel` is called, potentially leading to poor performance

Reply:
Done :-)